openedx/core/djangoapps/programs/utils.py

```diff
@@ -327,6 +328,11 @@ def supplement_program_data(program_data, user):
         program_data (dict): Representation of a program.
         user (User): The user whose enrollments to inspect.
     """
+    for organization in program_data['organizations']:
+        org_obj = get_organization_by_short_name(organization['key'])
```

Comment:
@schenedx for some pre-determined period, like an hour. Our API client wrapper function contains a good [example](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/edx_api_utils.py#L42-L71) of this kind of cache use. You could make the case that the cleanest way to implement this kind of caching is somewhere within the organization API itself, which is why we may want to hold off on doing it.